opentuner/search/technique.py

```diff
@@ -204,15 +223,20 @@ def call_main_generator(self):
                                     self.pending_tests)
         if self.pending_tests:
           self.rounds_since_novel_request = 0
-          yield None # wait
+          yield False # wait
 
 #list of all techniques
 the_registry = list()
 
+#list of composable techniques
+the_composable_registry = list()
 
 def register(t):
   the_registry.append(t)
 
+def register_composable(t):
```

Comment:
Just to clarify, we still use "register" for some composable technique instances. For instance, if I have instance a1, a2, a3 of composable class A, i could have 
register(a1)
register(a2)
register_composable(a2)
register_composable(a3)

This means that a user can directly use a1 or a2 (call --technique=a2) as a search technique but not a3. If a user tries to randomly generate a composable technique, the new technique could be generated from a2 or a3, but not a1.

With that out of the way, we wouldn't need to add a "register_model_driven" or "register_hill_climber" unless theres some specific functionality for getting available techniques to use that can only be done with the model-driven/hill-climber class of techniques?

I can't think of "technique getting" functionality we would support besides just getting a registered technique and getting a techniqe composed with operators.